tests/globalhelper/nodes.go

```diff
@@ -4,22 +4,78 @@ import (
 	"context"
 	"fmt"
 
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // EnableMasterScheduling enables/disables master nodes scheduling.
 func EnableMasterScheduling(scheduleable bool) error {
-	scheduler, err := GetAPIClient().OcpClientInterface.Schedulers().Get(context.TODO(), "cluster", metav1.GetOptions{})
+	// Get all nodes in the cluster
+	nodes, err := GetAPIClient().CoreV1Interface.Nodes().List(context.Background(), metav1.ListOptions{})
 	if err != nil {
-		return fmt.Errorf("failed to get schedulers: %w", err)
+		return fmt.Errorf("failed to get nodes: %w", err)
 	}
 
-	scheduler.Spec.MastersSchedulable = scheduleable
+	// Loop through the nodes and modify the taints
+	for _, node := range nodes.Items {
+		if isMasterNode(&node) {
+			if scheduleable {
+				err = removeControlPlaneTaint(&node)
+				if err != nil {
+					return fmt.Errorf("failed to set node %s schedulable value: %w", node.Name, err)
+				}
+			} else {
+				err = addControlPlaneTaint(&node)
+				if err != nil {
+					return fmt.Errorf("failed to set node %s schedulable value: %w", node.Name, err)
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func addControlPlaneTaint(node *corev1.Node) error {
+	// add the control-plane:NoSchedule taint to the master
+	node.Spec.Taints = append(node.Spec.Taints, corev1.Taint{
+		Key:    "node-role.kubernetes.io/control-plane",
+		Effect: corev1.TaintEffectNoSchedule,
+	})
+
+	_, err := GetAPIClient().CoreV1Interface.Nodes().Update(context.Background(), node, metav1.UpdateOptions{})
```

Comment:
is context.TODO() same as context.Background()? not sure.. Is the code waiting for this change to be applied somewhere else?

Reply:
Whoops, yes they are the exact same function.  I'll switch this to TODO().

Reply:
Actually there are a ton of `context.Background()` references in the code and I can do that in another PR to keep everything separate.